src/qibolab/platform/platform.py

```diff
@@ -226,7 +226,7 @@ def _execute(
     def execute(
         self,
         sequences: list[PulseSequence],
-        options: ExecutionParameters,
+        options: Optional[ExecutionParameters] = None,
```

Comment:
Great, this is making the call truly minimal!
```py
platform.execute([sequence, ...])
```

Reply:
Another option is to accept `ExecutionParameters` attributes, as `.execute()` arguments, either exclusively
```py
def execute(self, sequences, ..., **kwargs):
    options = ExecutionParameters(**kwargs)
```
or as a further option
```py
def execute(self, sequences, options = None, ..., **kwargs):
    if options is None:
        options = ExecutionParameters(**kwargs)
```

In case, I'd prefer the first alternative, since "there is only one way", and then it's less confusing (maybe it's not that intuitive to imagine that, in the second case, if you specify explicitly `options`, the other parameters will be ignored).

However, I'm not 100% sure it's just better than the current alternative. Though it would be often less redundant.

Reply:
Over the two, I would also prefer the first and I believe it also works in the case where no `kwargs` are given. 

From a usability point of view it is probably simpler, since we don't even need to expose `ExecutionParameters` itself in the public API (one less thing for the user to search for/remember).